client/src/pages/proverb/Proverb.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect } from "react";
+import { Form } from "react-bootstrap";
+import Input from "./../../components/Input";
+import Button from "./../../components/Button";
+import { isArabic } from "./../../helpers/functions";
+
+import { PROVERB_INITIAL_DATA } from "./../../helpers/formData";
+import "./Proverb.css";
+
+const Proverb = ({ actionType }) => {
+  const [formData, setFormData] = useState(PROVERB_INITIAL_DATA);
+  const [disabled, setDisabled] = useState(true);
+
+  const { proverb, translation, explanation } = formData;
+
+  useEffect(() => {
+    const isFilled = [proverb, translation, explanation].every((data) =>
+      Boolean(data)
+    );
+    const result = isArabic(proverb);
+    isFilled && result ? setDisabled(false) : setDisabled(true);
+  }, [formData]);
+
+  const onChange = (e) => {
```

Comment:
```suggestion
  const handleInputChange = (e) => {
```

Lets make this function name more explicit as well, to make it even more readable :)